SOLR-14920: Spotless formatting for test-framework

risdenk opened the pull request:
https://issues.apache.org/jira/browse/SOLR-14920

risdenk commented on solr/test-framework/src/test/org/apache/solr/TestLogLevelAnnotations.java:
```diff
@@ -27,36 +27,35 @@
+   * evaluated -- modifying the log4j run time configuration. There is no reason why the <code>
+   * @LogLevel</code> configuration of this class <em>should</em> affect the "root" Logger, but
+   * setting this in static class initialization protect us (as best we can) against the possibility
+   * that it <em>might</em> due to an unforseen (future) bug.
```
Fix this

risdenk commented on solr/test-framework/src/test/org/apache/solr/TestLogLevelAnnotations.java:
```diff
@@ -65,31 +64,38 @@ public static void checkLogLevelsBeforeClass() {
+   * <p><b>NOTE:</b> We only validate <code>@LogLevel</code> modifications made at the {@link
+   * #testMethodLogLevels} level, not at the 'class' level, because of the lifecycle of junit
+   * methods: This <code>@AfterClass</code> will run before the <code>SolrTestCaseJ4</code> <code>
+   * @AfterClass</code> method where the 'class' <code>@LogLevel</code> modifications will be reset.
```
Fix this

risdenk commented on solr/test-framework/src/java/org/apache/solr/BaseDistributedSearchTestCase.java:
```diff
@@ -222,14 +219,15 @@ protected BaseDistributedSearchTestCase() {
 
-  private final static int DEFAULT_MAX_SHARD_COUNT = 3;
+  private static final int DEFAULT_MAX_SHARD_COUNT = 3;
+
+  private int shardCount =
+      -1; // the actual number of solr cores that will be created in the cluster
 
-  private int shardCount = -1;      // the actual number of solr cores that will be created in the cluster
```
Fix this

risdenk commented on solr/test-framework/src/java/org/apache/solr/SolrTestCaseJ4.java:
```diff
@@ -2216,32 +2338,40 @@ public static void copyXmlToHome(File dstRoot, String fromFile) throws IOExcepti
-  // Creates a consistent configuration, _including_ solr.xml at dstRoot. Creates collection1/conf and copies
+  // Creates a consistent configuration, _including_ solr.xml at dstRoot. Creates collection1/conf
+  // and copies
   // the stock files in there.
```
Fix this

risdenk commented on solr/test-framework/src/java/org/apache/solr/SolrTestCaseJ4.java:
```diff
@@ -3022,26 +3180,26 @@ private static void randomizeNumericTypesProperties() {
   /**
-   * Cleans up the randomized sysproperties and variables set by {@link #randomizeNumericTypesProperties}
+   * Cleans up the randomized sysproperties and variables set by {@link
+   * #randomizeNumericTypesProperties}
```
Fix this

risdenk reviewed (commented):
Went through all the files quickly and highlighted a few things to address

dsmiley commented on solr/test-framework/src/java/org/apache/solr/SolrJettyTestBase.java:
```diff
@@ -160,8 +165,10 @@ public SolrClient createNewSolrClient() {
 
-  // Sets up the necessary config files for Jetty. At least some tests require that the solrconfig from the test
-  // file directory are used, but some also require that the solr.xml file be explicitly there as of SOLR-4817
+  // Sets up the necessary config files for Jetty. At least some tests require that the solrconfig
```
comment style here doesn't flow well as Javadoc; maybe convert to Javavdoc, or reflow manually

dsmiley commented on solr/test-framework/src/java/org/apache/solr/SolrTestCaseHS.java:
```diff
@@ -107,72 +106,84 @@ public static ModifiableSolrParams params(SolrParams params, String... moreParam
-          "\n\texpected={}\n\tmodel={}"
-          , err, req, strResponse, JSONUtil.toJSON(model), fullModel);
+      log.error(
+          "RESPONSE MISMATCH: {}\n\trequest={}\n\tresult={}" + "\n\texpected={}\n\tmodel={}",
```
I bet if you declare the first arg as a String in the previous line (`var msg = "RESPONSE ......."`) then it would then the following log.error would be one line.

dsmiley commented on solr/test-framework/src/java/org/apache/solr/SolrTestCaseHS.java:
```diff
@@ -181,53 +192,56 @@ public static void matchJSON(String response, String... tests) throws Exception
+   * SolrQueryRequest req = req();
+   * req.getSearcher().getnCache().clear();  // OFF-HEAP
+   * req.close();
+   * }***/
```
yuck

dsmiley commented on solr/test-framework/src/java/org/apache/solr/SolrTestCaseJ4.java:
```diff
@@ -164,44 +171,36 @@
 @SuppressSysoutChecks(bugUrl = "Solr dumps tons of logs to console.")
-@SuppressFileSystems("ExtrasFS") // might be ok, the failures with e.g. nightly runs might be "normal"
+@SuppressFileSystems(
+    "ExtrasFS") // might be ok, the failures with e.g. nightly runs might be "normal"
```
Move the comment to the line before, or shorten it

dsmiley commented on solr/test-framework/src/java/org/apache/solr/SolrTestCaseJ4.java:
```diff
@@ -1849,73 +1914,90 @@ public Fld createField() {
+    String responseStr =
+        h.query(
+            req(
+                "q",
```
ok for now but if I were writing fresh code and saw spotless do this, I would go about writing the code differently.  I'd call `params().set("q", "*:*").set(...).set(...)` etc.

dsmiley commented on solr/test-framework/src/java/org/apache/solr/SolrTestCaseJ4.java:
```diff
@@ -2147,34 +2247,51 @@ public static void assertXmlFile(final File file, String... xpath)
       if (null != results) {
-        String msg = "File XPath failure: file=" + file.getPath() + " xpath="
-            + results + "\n\nxml was: " + xml;
+        String msg =
```
minor: I suspect inlining msg to the `fail(....)` would be _just_ short enough to not overflow

risdenk replied:
Checked and its still too long.

dsmiley commented on solr/test-framework/src/java/org/apache/solr/SolrTestCaseJ4.java:
```diff
@@ -2877,46 +3010,63 @@ protected String getSaferTestName() {
+    assertFalse(
+        "SecureRandom algorithm '"
+            + algorithm
+            + "' is in use by your JVM, "
```
I wish spotless would recognize how to break these strings in a way that leverages the full line width

dsmiley commented on solr/test-framework/src/java/org/apache/solr/cloud/AbstractBasicDistributedZk2TestBase.java:
```diff
@@ -41,28 +40,31 @@
-@LuceneTestCase.SuppressCodecs({"SimpleText"}) // Backups do checksum validation against a footer value not present in 'SimpleText'
+@LuceneTestCase.SuppressCodecs({
+  "SimpleText"
+}) // Backups do checksum validation against a footer value not present in 'SimpleText'
```
Moving the comment to a line before the annotation would be more succinct

dsmiley commented on solr/test-framework/src/java/org/apache/solr/cloud/AbstractFullDistribZkTestBase.java:
```diff
@@ -257,16 +260,27 @@ public void distribSetUp() throws Exception {
       System.clearProperty(URL_SCHEME);
-      try (ZkStateReader zkStateReader = new ZkStateReader(zkServer.getZkAddress(),
-          AbstractZkTestCase.TIMEOUT, AbstractZkTestCase.TIMEOUT)) {
+      try (ZkStateReader zkStateReader =
```
an aside: perfect spot to use "var" -- the variable's name makes the type redundant.  And it's shorter.

dsmiley commented on solr/test-framework/src/java/org/apache/solr/cloud/SolrCloudTestCase.java:
```diff
@@ -72,37 +72,42 @@
 
-  public static final int DEFAULT_TIMEOUT = 45; // this is an important timeout for test stability - can't be too short
+  public static final int DEFAULT_TIMEOUT =
+      45; // this is an important timeout for test stability - can't be too short
```
move comment to previous line

dsmiley commented on solr/test-framework/src/java/org/apache/solr/cloud/SolrCloudTestCase.java:
```diff
@@ -72,37 +72,42 @@
-    // to use (distributed or Overseer based) and which cluster update strategy to use (distributed if collection API
-    // is distributed, but Overseer based or distributed randomly chosen if Collection API is Overseer based)
+    // By default the MiniSolrCloudCluster being built will randomly (seed based) decide which
+    // collection API strategy
```
reflow; maybe use `/* ... */` style and perhaps spotless might even do for you?

dsmiley commented on solr/test-framework/src/java/org/apache/solr/util/LogLevel.java:
```diff
@@ -81,33 +77,33 @@ public static void restoreLogLevels(Map<String, Level> savedLogLevels) {
+              }
+            } else {
+              // there is no existing configuration for the exact loggerName, logConfig is some
+              // ancestor
```
reflow

dsmiley commented on solr/test-framework/src/java/org/apache/solr/util/LogListener.java:
```diff
@@ -326,26 +351,31 @@ public String pollMessage() {
-    // This could probably be implemented with a combination of "LevelMatchFilter" and "ConjunctionFilter" if "ConjunctionFilter" existed
-    // Since it doesn't, we write our own more specialized impl instead of writing & combining multiple generalized versions
+    // This could probably be implemented with a combination of "LevelMatchFilter" and
+    // "ConjunctionFilter" if "ConjunctionFilter" existed
```
reflow

dsmiley commented on solr/test-framework/src/test/org/apache/solr/cloud/MiniSolrCloudClusterTest.java:
```diff
@@ -101,66 +100,85 @@ public void testExtraFilters() throws Exception {
-    // regardless of what sys prop may be set, everything in the cluster should use solr home dirs under the 
-    // configured base dir -- and nothing in the call stack should be "setting" the sys prop to make that work...
+    // regardless of what sys prop may be set, everything in the cluster should use solr home dirs
+    // under the
```
reflow

dsmiley reviewed (commented):
mostly just some reflow issues.  I wonder if spotless will reflow `/* foo bar */` style comments.

risdenk commented:
@dsmiley  pretty sure I addressed all the comments I could in [261d936](https://github.com/apache/solr/pull/704/commits/261d936f1afede6f4b4c0abd1f70025361147a51)
